core/src/main/java/com/digitalpebble/stormcrawler/parse/JSoupFilter.java

```diff
@@ -21,7 +21,7 @@
  * content. They are used exclusively by {@link
  * com.digitalpebble.stormcrawler.bolt.JSoupParserBolt}.
  */
-public abstract class JSoupFilter implements Configurable {
+public interface JSoupFilter extends Configurable {
```

Comment:
I would rather keep it as an abstract class.

Reply:
> Have gone through the changes. As discussed previously, it is neither practical nor reassuring to have 109 files changes in one go: spotting a problem in so many modifications is rather difficult. Could we please not have that in the future. Some points are still open for discussion (see comments). Some changes need to be reverted. Thanks!

`Could we please not have that in the future.`: Definetly. I'm sorry to putting you through that. Future PRs will be in a reasonable size. Thank you for putting up with this PR.

I'll revert the changes you marked when I find some time to do it.

Reply:
That class declares only a single abstract function with no further logic. Declaring it as an interface would not change much but give more possibilities for inheritance/implementation.

A possible use-case is that you could use a configurable base-class that manages the connection for some kind of database for implementing a custom JSoupFilter.

Reply:
Okay, forget what I said about the use case. You could get the same result by using a object for that.

Are you planning on extending that JsoupFilter at some point? Or is it just your personal taste?
I'm okay with both, but I personally think that an "abstract only" class should always be an interface.

Reply:
The class could possibly extended in the future + as a personal preference, I find abstract classes more future-proof.

Reply:
Ok, I'll revert it.